website/content/en/docs/reference/configuration/unit-tests.md

```diff
@@ -0,0 +1,359 @@
+
+[[tests]]
+name = "Test add_metadata transform"
+
+[[tests.inputs]]
+insert_at = "add_metadata"
+
+[tests.inputs.log_fields]
+message = "<102>1 2020-12-22T15:22:31.111Z vector-user.biz su 2666 ID389 - Something went wrong"
+tags.environment = "production"
+```
+
+### Outputs
+
+In the `outputs` array of your unit testing configuration you specify two things:
+
+Parameter | Type | Description
+:---------|:-----|:-----------
+`extract_from` | string (name of transform) | The transform whose output you want to test.
+`conditions` | array of objects | The [VRL conditions][verifying] to run against the output.
+
+Each condition in the `conditions` array has two fields:
+
+Parameter | Type | Description
+:---------|:-----|:-----------
+`type` | string | The type of condition you're providing. As the original `check_fields` syntax is now deprecated, this defaults to  `vrl`.
+`source` | string (VRL Boolean expression) | Explained in detail [above](#verifying).
+
+
+both the expected output
+events from the transform(s) you specified in in the [`inputs`](#inputs) array as well the point in
+the transform chain from which output events are to be extracted.
+
+Here's an example `outputs` declaration:
+
+```toml
+[[tests.outputs]]
+extract_from = "add_metadata"
+
+[[tests.outputs.conditions]]
+type = "vrl"
+source = '''
+is_string(.id) && exists(.tags)
+'''
+```
+
+
+{{< danger title="`check_fields` conditions now deprecated" >}}
+Vector initially provided a `check_fields` condition type that enabled you to specify Boolean
+test conditions using a special configuration-based system. `check_fields` is now deprecated. We
+strongly recommend converting any existing `check_fields` tests to `vrl` conditions.
+{{< /danger >}}
+
+### Event types
+
+There are currently two event types that you can unit test in Vector:
+
+* [`log`](#logs) events
+* [`metric`](#metrics) events
+
+#### Logs
+
+As explained in the section on [inputs](#inputs) above, when testing log events you have can specify
+either a structured event [object] or a raw [string].
+
+##### Object
+
+To specify a structured log event as your test input, use `log_fields`:
+
+```toml
+[tests.inputs.log_fields]
+message = "successful transaction"
+code = 200
+id = "38c5b0d0-5e7e-42aa-ae86-2b642ad2d1b8"
+```
+
+##### Raw string value
+
+To specify a raw string value for a log event, use `value`:
+
+```toml
+[[tests.inputs]]
+insert_at = "add_metadata"
+value = "<102>1 2020-12-22T15:22:31.111Z vector-user.biz su 2666 ID389 - Something went wrong"
+```
+
+#### Metrics
+
+You can specify the fields in a metric event to be unit tested using a `metric` object:
+
+```toml
+[[tests.inputs]]
+insert_at = "my_metric_transform"
+type = "metric"
+
+[tests.inputs.metric]
+name = "count"
+kind = "absolute"
+counter.value = 1
+```
+
+Here's a full end-to-end example of unit testing a metric through a transform:
+
+```toml
+[transforms.add_unique_id_to_metric]
+type = "remap"
+inputs = []
+source = '''
+.id = uuid_v4()
```

Comment:
I thought (based on what I remember from last week, which is very little) this wanted to be `.tags.id`

Reply:
Also as an aside - seems like a weird thing to do with metrics due to cardinality... maybe removing tags or setting a tag based on the vector hostname may be more realistic

Reply:
Yeah, true. A real world example we heard recently was converting statsd metrics, which cannot have tags, by parsing the name like coverting:

```
expedia.some_svc.region.emea.datacenter.de2.api.request_latency 1
```

into

```
expedia.some_svc.api.request_latency{region=emea,datacenter=de2} 1
```

This could be done via regex parsing.

Reply:
Yeah, true. A real world example we heard recently was converting statsd metrics, which cannot have tags, by parsing the name like coverting:

```
foo.some_svc.region.emea.datacenter.de2.api.request_latency 1
```

into

```
foo.some_svc.api.request_latency{region=emea,datacenter=de2} 1
```

This could be done via regex parsing.

Reply:
@jszwedko Ah yes, good call. I am not awesome at examples.